geom/encoding/geojson/geojson.go

```diff
@@ -117,3 +127,23 @@ type FeatureCollection struct {
 	Type     featureCollectionType `json:"type"`
 	Features []Feature             `json:"features"`
 }
+
+// closePolygon will check if the polygon has the same value for the first and last point. if the value
+// is not the same it will add an extra point to "close" the polygon per the GeoJSON spec.
+func closePolygon(p geom.Polygoner) geom.Polygoner {
+	rings := p.LinearRings()
+	for i := range rings {
+
+		if len(rings[i]) < 3 {
+			return geom.Polygon(rings)
+		}
+
+		//	check if the first point and the last point are the same
+		//	if they're not, make a copy of the first point and add it as the last position
+		if rings[i][0] != rings[i][len(rings[i])-1] {
+			rings[i] = append(rings[i], rings[i][0])
+		}
+	}
+
+	return geom.Polygon(rings)
+}
```

Comment:
We don't want to abandon all the rings after rings with less then 3 points. I think you should return a new slice.
```
fun closePolygon(p geom.Polygoner) geom.Polygoner {
      rings := p.LinearRings()
       r = make([][][2]float64, 0, len(rings))
      for i := range rings {
                   if len(rings[i]) < 3 { 
                           continue
                  }
                  // check if the first point and the last point are the same, if not make them.
                  if rings[i][0] != rings[i][len(rings[i])-1] {
                       r = append(r, append(rings[i],rings[i][0]))
                  } else {
                      r = append(r, rings[i])
                 }
            }
           return geom.Polygon(r)
}
```

The way this is done should have the same memory foot print as what you had before.

Reply:
:+1: Plus log an error or warning that you've found an invalid polygon & are skipping it.